catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/SecureWebSocketServlet.java

```diff
@@ -90,15 +108,23 @@ public void onError(Session session, Throwable ex) {
 
     @OnWebSocketMessage
     public void onMessage(Session session, String message) {
-      runWithUser(
-          session,
-          () -> {
-            try {
-              ws.onMessage(session, message);
-            } catch (IOException e) {
-              LOGGER.error("Failed to receive ws message.", e);
-            }
-          });
+      if (isUserLoggedIn()) {
+        runWithUser(
+            session,
+            () -> {
+              try {
+                ws.onMessage(session, message);
+              } catch (IOException e) {
+                LOGGER.error("Failed to receive ws message.", e);
+              }
+            });
+      } else {
+        onError(session, new WebSocketAuthenticationException("User not logged in.", message));
+      }
+    }
+
+    private boolean isUserLoggedIn() {
+      return securityTokenHolder.getRealmTokenMap().size() > 0;
```

Comment:
So we are checking if a user is logged in by checking the `securityTokenHolder`. And it looks like the `securityTokenHolder` is created when the `configure()` method is called.

❓ Is this a possible case: `configure()` is called on this servlet, the `securityTokenHolder` in this class is instantiated with a valid security token, some time passes (say 10 seconds) and the security token expires. Would this servlet then still assume the user is logged in because the `securityTokenHolder` contains the now invalid token?

Reply:
The `configure()` method is only called once in the whole websocket session. When the security token expires, the token is removed from the tokenholder, so that's why this checks for the token being in the token holder.

Reply:
Ahhh 👍 , `req.getSession().getAttribute(SecurityConstants.SAML_ASSERTION)` is being **cast** to a SecurityTokenHolder, not **constructed** into one. I thought we were making our own token holder and adding the token to it each time, instead of holding a reference to the ... "main" ... token holder, which will pick up changes in the system